src/player_activity.cpp

```diff
@@ -318,6 +318,11 @@ void player_activity::finish( player *p )
             p->do_read(&(p->i_at(position)));
             if (type == ACT_NULL) {
                 add_msg(_("You finish reading."));
+
+                // below code added in to support reading from ground/vehicles
+                if ( values[1] != FRM_INV ) {
+                    g->drop(p->inv.size()-1);
```

Comment:
`position` is equal to `p->inv.size() - 1` at the time the action started.
If even one item is removed (and not replaced) from the inventory, `p->i_at(position)` will return a null item. This will cause a crash as null item is not a book.

Though yeah, they should both use the same index - whether `position` or `inv.size() - 1`.